From c71d2a64892acb652163c7223e6113b4524063ab Mon Sep 17 00:00:00 2001
From: Geert Uytterhoeven <geert@linux-m68k.org>
Date: Thu, 5 Feb 2009 22:06:03 +0100
Subject: [PATCH 39/90] m68k: atari - Rename "mfp" to "st_mfp"

http://kisskb.ellerman.id.au/kisskb/buildresult/72115/:
| net/mac80211/ieee80211_i.h:327: error: syntax error before 'volatile'
| net/mac80211/ieee80211_i.h:350: error: syntax error before '}' token
| net/mac80211/ieee80211_i.h:455: error: field 'sta' has incomplete type
| distcc[19430] ERROR: compile net/mac80211/main.c on sprygo/32 failed

This is caused by

| # define mfp ((*(volatile struct MFP*)MFP_BAS))

in arch/m68k/include/asm/atarihw.h, which conflicts with the new "mfp" enum in
net/mac80211/ieee80211_i.h.

Rename "mfp" to "st_mfp", as it's a way too generic name for a global #define.

Signed-off-by: Geert Uytterhoeven <geert@linux-m68k.org>
---
 drivers/char/atari_scc.c     |    4 ++--
 drivers/net/atari_91C111.c   |    4 ++--
 drivers/net/atari_ethernec.c |    4 ++--
 3 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/drivers/char/atari_scc.c b/drivers/char/atari_scc.c
index 23fc4ed..2bbd44c 100644
--- a/drivers/char/atari_scc.c
+++ b/drivers/char/atari_scc.c
@@ -653,7 +653,7 @@ int atari_scc_init(void)
 	if (!(ATARIHW_PRESENT(SCC) || ATARIHW_PRESENT(ST_ESCC)))
 		return -ENODEV;
 
-	scc_del = &mfp.par_dt_reg;
+	scc_del = &st_mfp.par_dt_reg;
 
 	if (MACH_IS_TT)
 		res = atari_tt_scc_init();
@@ -1541,7 +1541,7 @@ static void atari_init_scc_port(int cflag)
 	int baud = cflag & CBAUD;
 	int clksrc, clkmode, div, reg3, reg5;
 
-	scc_del = &mfp.par_dt_reg;
+	scc_del = &st_mfp.par_dt_reg;
 
 	if (cflag & CBAUDEX)
 		baud += B38400;
diff --git a/drivers/net/atari_91C111.c b/drivers/net/atari_91C111.c
index af2d255..a82ba3d 100644
--- a/drivers/net/atari_91C111.c
+++ b/drivers/net/atari_91C111.c
@@ -2302,9 +2302,9 @@ static int __devinit smc_drv_probe(struct platform_device *pdev)
 			timd = 255;
 		printk(KERN_INFO "Timer D frequency: %u Hz\n", 38400/timd);
 		/* set Timer D data Register */
-		mfp.tim_dt_d = timd;	/* 200 Hz */
+		st_mfp.tim_dt_d = timd;	/* 200 Hz */
 		/* start timer D, div = 1:100 */
-		mfp.tim_ct_cd = (mfp.tim_ct_cd & 0xf0) | 0x5;
+		st_mfp.tim_ct_cd = (st_mfp.tim_ct_cd & 0xf0) | 0x5;
 	}
 
 	return 0;
diff --git a/drivers/net/atari_ethernec.c b/drivers/net/atari_ethernec.c
index 74eeae4..05b5e2e 100644
--- a/drivers/net/atari_ethernec.c
+++ b/drivers/net/atari_ethernec.c
@@ -622,9 +622,9 @@ static int __init ne_probe1(struct net_device *dev, int ioaddr)
 		/* timer routine set up in atari_ethernec_probe() */
 		if (dev->irq == IRQ_MFP_TIMD) {
 			/* set Timer D data Register */
-			mfp.tim_dt_d = 123;	/* 200 Hz */
+			st_mfp.tim_dt_d = 123;	/* 200 Hz */
 			/* start timer D, div = 1:100 */
-			mfp.tim_ct_cd = (mfp.tim_ct_cd & 0xf0) | 0x6;
+			st_mfp.tim_ct_cd = (st_mfp.tim_ct_cd & 0xf0) | 0x6;
 		}
 		/* Must make this shared in case other timer ints are needed */
 		ret = request_irq(dev->irq, atari_ei_interrupt, IRQF_SHARED, name, dev);
-- 
1.6.2.3

